pyface/ui/wx/action/action_item.py

```diff
@@ -122,30 +122,22 @@ def __init__(self, parent, menu, item, controller):
 
         # Listen for trait changes on the action (so that we can update its
         # enabled/disabled/checked state etc).
-        action.on_trait_change(self._on_action_enabled_changed, "enabled")
-        action.on_trait_change(self._on_action_visible_changed, "visible")
-        action.on_trait_change(self._on_action_checked_changed, "checked")
-        action.on_trait_change(self._on_action_name_changed, "name")
-        action.on_trait_change(self._on_action_image_changed, "image")
+        action.observe(self._on_action_enabled_changed, "enabled")
+        action.observe(self._on_action_visible_changed, "visible")
+        action.observe(self._on_action_checked_changed, "checked")
+        action.observe(self._on_action_name_changed, "name")
+        action.observe(self._on_action_image_changed, "image")
 
         if controller is not None:
             self.controller = controller
             controller.add_to_menu(self)
 
     def dispose(self):
         action = self.item.action
-        action.on_trait_change(
-            self._on_action_enabled_changed, "enabled", remove=True
-        )
-        action.on_trait_change(
-            self._on_action_visible_changed, "visible", remove=True
-        )
-        action.on_trait_change(
-            self._on_action_checked_changed, "checked", remove=True
-        )
-        action.on_trait_change(
-            self._on_action_name_changed, "name", remove=True
-        )
+        action.observe(self._on_action_enabled_changed, "enabled", remove=True)
+        action.observe(self._on_action_visible_changed, "visible", remove=True)
+        action.observe(self._on_action_checked_changed, "checked", remove=True)
+        action.observe(self._on_action_name_changed, "name", remove=True)
```

Comment:
For some reason we don't remove the listener for "image" here...? I left code to match it as is but this feels like a bug?  Additionally for some of the occurrences below, we do not remove

Reply:
Note for future ref: Maybe the segfault was related to this?

Reply:
Yep, definitely should be removing anything that is added.